### PR TITLE
WordPress - check version from custom file

### PR DESCRIPTION
**Pull request by firefart**
This introduces a new option to check a custom file for versions.

Needed for #5290

### CODE REVIEW

**Comment by TomSellers** [comment]
This works like a champ for my purposes and is much cleaner.  Win all around.  Thanks much!

**Comment by espreto** [comment]
@FireFart That's really good. Good point. :+1:

**Comment by wvu-r7** [comment]
YARD \*and\* RSpec? We have a winner!

File: lib/msf/http/wordpress/version.rb [hunk 1]
```diff
@@ -82,6 +82,26 @@ def check_theme_version_from_readme(theme_name, fixed_version = nil, vuln_introd
...
+  def check_version_from_custom_file(uripath, regex, fixed_version = nil, vuln_introduced_version = nil)
+    res = send_request_cgi(
+      'uri'    => uripath,
+      'method' => 'GET'
+    )
+
+    # file not found
+    return Msf::Exploit::CheckCode::Unknown if res.nil? || res.code != 200
+
+    return extract_and_check_version(res.body.to_s, :custom, 'custom file', fixed_version, vuln_introduced_version, regex)
```
Comment (wvu-r7) [inline lib/msf/http/wordpress/version.rb]: An explicit ```return``` isn't necessary.

File: lib/msf/http/wordpress/version.rb [hunk 2]
```diff
@@ -137,7 +157,7 @@ def check_version_from_readme(type, name, fixed_version = nil, vuln_introduced_v
     end
   end
 
-  def extract_and_check_version(body, type, item_type, fixed_version = nil, vuln_introduced_version = nil)
+  def extract_and_check_version(body, type, item_type, fixed_version = nil, vuln_introduced_version = nil, regex = nil)
```
Comment (wvu-r7) [inline lib/msf/http/wordpress/version.rb]: Might be a good idea to start using options hashes or keyword arguments.

**Comment by wvu-r7** [comment]
https://github.com/rapid7/metasploit-framework/commit/18791ce93340322ff69f25b1e65795172036c7f3

**Comment by wvu-r7** [comment]
Bitte!

**Pull request merged**